### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by abesto**
As part of https://github.com/openzipkin/zipkin/issues/2102

Used https://github.com/openzipkin/zipkin-aws/pull/95 as a template; `diff` between the 1.0 `circle.yml` files:

```diff
# diff zipkin-{aws,azure}/circle.yml 
2c2
< # Copyright 2016-2017 The OpenZipkin Authors
---
> # Copyright 2017 The OpenZipkin Authors
16,17c16,17
<   environment:
<     JAVA_HOME: /usr/lib/jvm/zulu-9-amd64
---
>   java:
>     version: openjdk8
21,24d20
<     - sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0x219BD9C9
<     - sudo apt-add-repository -y "deb http://repos.azulsystems.com/ubuntu stable main"
<     - sudo apt-get -y update
<     - sudo apt-get -y install zulu-9
```

So essentially the same, except for the copyright year and that `zipkin-azure` is on JDK 8.

`diff` of 2.0 `.circleci/config.yml` after this PR:

```diff
# diff zipkin-{aws,azure}/.circleci/config.yml
2c2
< # Copyright 2016-2018 The OpenZipkin Authors
---
> # Copyright 2017-2018 The OpenZipkin Authors
19c19
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
51c51
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
66c66
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
81c81
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          command: |
+            mkdir -p /tmp/test-reports/unit-tests/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} /tmp/test-reports/unit-tests/ \;
+
+      - store_test_results:
+          path: /tmp/test-reports/
+
+  publish_snapshot:
+    docker:
+      - image: circleci/openjdk:9-jdk
```
Comment (shakuzen) [inline .circleci/config.yml]: 8 instead of 9?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          name: Publish snapshot
+          command: |
+            # Guard against running this on pull requests or forks
+            [ "$CIRCLE_PROJECT_USERNAME" == "openzipkin" ] || exit 0
+            [ -z "$CIRCLE_PR_NUMBER" ] || exit 0
+            ./build-support/publish-snapshot.sh
+
+  publish_stable:
+    docker:
+      - image: circleci/openjdk:9-jdk
```
Comment (shakuzen) [inline .circleci/config.yml]: 8 instead of 9?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+          name: Publish stable version
+          command: |
+            # Guard against running this on pull requests or forks
+            [ "$CIRCLE_PROJECT_USERNAME" == "openzipkin" ] || exit 0
+            [ -z "$CIRCLE_PR_NUMBER" ] || exit 0
+            ./build-support/publish-stable.sh
+
+  trigger_publish:
+    docker:
+      - image: circleci/openjdk:9-jdk
```
Comment (shakuzen) [inline .circleci/config.yml]: 8 instead of 9?

File: .circleci/config.yml (hunk 1 above)
Comment (abesto) [reply .circleci/config.yml]: Bah. Thanks.

**Comment by abesto** [comment]
Good thought on factoring out the docker image. I won't include that in this PR to preserve consistency across the CircleCI-built repos; sounds like a good improvement if we decide to stay with Circle / use it more long-term.

**Comment by adriancole** [comment]
good start. we will need to switch to 9 soon, but a straight port makes sense now.

**Pull request merged**